### PR TITLE
Refactor index editor

### PR DESCRIPTION
**Pull request by bheni**
This takes the same approach to the index editor that was used with the table editor. It also has to thread a bunch of stuff through in order to create the appropriate editors in different situations.

### CODE REVIEW

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Review by Hydrocharged — APPROVED** [review]
LGTM! Just some small comments

File: go/libraries/doltcore/table/editor/table_edit_session.go [hunk 1]
```diff
@@ -28,24 +29,35 @@ import (
...
 
-// TableEditSessionProps are properties that define different functionality for the TableEditSession.
-type TableEditSessionProps struct {
+// Options are properties that define different functionality for the TableEditSession.
+type Options struct {
 	ForeignKeyChecksDisabled bool // If true, then ALL foreign key checks AND updates (through CASCADE, etc.) are skipped
+	Deaf                     DbEaFactory
+}
+
+func TestEditorOptions(vrw types.ValueReadWriter) Options {
```
[the remark this answers is not in the document]
Comment (bheni) [reply go/libraries/doltcore/table/editor/table_edit_session.go:45]: I wish it was possible.  But go won't let you put exported functions in test files to be used in the test files of other packages.

**Pull request merged**